service/grails-app/services/org/olf/rs/statemodel/actions/iso18626/ActionISO18626RequesterService.groovy

```diff
@@ -39,9 +40,13 @@ public abstract class ActionISO18626RequesterService extends ActionISO18626Servi
         String note = sequenceResult.note;
         request.lastSequenceReceived = sequenceResult.sequence;
 
+        if (EventMessageRequestIndService.isSlnpRequesterStateModel(request)) {
+            Set<RequestVolume> volumes = new HashSet<>(request.volumes)
+            volumes.forEach {it -> request.removeFromVolumes(it)}
+        }
 
         // if parameters.deliveryInfo.itemId then we should stash the item id
-        if (parameters?.deliveryInfo) {
+        if (parameters?.deliveryInfo) { // TODO check if status is Loaned or CopyCompleted
```

Comment:
@JanisSaldabols itemId handling to reset volumes is missing below.

Reply:
It is below 
I just added that for SLNP we always remove existing volumes

Reply:
Ok, is this code only used for StatusChange or also when creating new or retrying request?